### PR TITLE
chore: seal internal classes when possible

### PR DESCRIPTION
**Pull request by Evangelink**

### CODE REVIEW

File: src/Platform/Microsoft.Testing.Platform/Hosts/TestHostBuilder.cs [hunk 1]
```diff
@@ -657,16 +656,13 @@ private async Task<ITestFramework> BuildTestFrameworkAsync(TestFrameworkBuilderD
...
-            await RegisterAsServiceOrConsumerOrBothAsync(new TestFrameworkProxy(testFramework), serviceProvider, dataConsumersBuilder);
+            await TestHostBuilder.RegisterAsServiceOrConsumerOrBothAsync(new TestFrameworkProxy(testFramework), serviceProvider, dataConsumersBuilder);
         }
         finally
         {
             serviceProvider.AllowTestAdapterFrameworkRegistration = false;
         }
 
-        // Virtual callback that allows to the VSTest mode to register custom services needed by the bridge.
-        AfterTestAdapterCreation(serviceProvider);
```
Comment (Youssef1313) [inline src/Platform/Microsoft.Testing.Platform/Hosts/TestHostBuilder.cs:668]: Not following with this change.

Comment (Youssef1313) [reply src/Platform/Microsoft.Testing.Platform/Hosts/TestHostBuilder.cs:668]: Okay I see now :)

File: src/Platform/Microsoft.Testing.Platform/ServerMode/JsonRpc/Json/Jsonite/Jsonite.cs [hunk 2]
```diff
@@ -771,7 +771,7 @@ private static bool IsHex(char c) =>
         }
 
         /// <summary>
-        /// The internal class used to serialize an object graph to a json text.
+        /// The internal sealed class used to serialize an object graph to a json text.
```
Comment (Youssef1313) [inline src/Platform/Microsoft.Testing.Platform/ServerMode/JsonRpc/Json/Jsonite/Jsonite.cs]: Unintentional?

File: src/Platform/Microsoft.Testing.Extensions.MSBuild/Serializers/FailedTestInfoRequestSerializer.cs
Comment (Youssef1313) [inline src/Platform/Microsoft.Testing.Extensions.MSBuild/Serializers/FailedTestInfoRequestSerializer.cs:1]: Consider sealing internal `record`s as well

**Review by Youssef1313 — DISMISSED** [review]
LGTM with two small comments

File: src/Platform/Microsoft.Testing.Platform/ServerMode/JsonRpc/Json/Jsonite/Jsonite.cs (hunk 2 above)
Comment (Evangelink) [inline src/Platform/Microsoft.Testing.Platform/ServerMode/JsonRpc/Json/Jsonite/Jsonite.cs]:
```suggestion
        /// The internal class used to serialize an object graph to a json text.
```

**Review by Youssef1313 — APPROVED** [review]

**Pull request merged**